cns/hnsclient/hnsclient_windows.go

```diff
@@ -153,3 +201,454 @@ func deleteHnsNetwork(networkName string) error {
 
 	return err
 }
+
+func configureHostNCApipaNetwork(localIPConfiguration cns.IPConfiguration) (*hcn.HostComputeNetwork, error) {
+	network := &hcn.HostComputeNetwork{
+		Name: hostNCApipaNetworkName,
+		Ipams: []hcn.Ipam{
+			hcn.Ipam{
+				Type: hcnIpamTypeStatic,
+			},
+		},
+		SchemaVersion: hcn.SchemaVersion{
+			Major: hcnSchemaVersionMajor,
+			Minor: hcnSchemaVersionMinor,
+		},
+		Type:  hostNCApipaNetworkType,
+		Flags: hcn.EnableNonPersistent, // Set up the network in non-persistent mode
+	}
+
+	if netAdapterNamePolicy, err := policy.GetHcnNetAdapterPolicy(hostNCLoopbackAdapterName); err == nil {
+		network.Policies = append(network.Policies, netAdapterNamePolicy)
+	} else {
+		return nil, fmt.Errorf("Failed to serialize network adapter policy. Error: %v", err)
+	}
+
+	// Calculate subnet prefix
+	// Following code calculates the subnet prefix from localIPConfiguration IP
+	// e.g. IP: 169.254.128.7 Prefix length: 17 then resulting subnet prefix: 169.254.128.0/17
+	// subnetPrefix: ffff8000
+	// subnetPrefix.IP: 169.254.128.0
```

Comment:
add a todo with workitem

Reply:
Added the workitem here:
https://msazure.visualstudio.com/One/_workitems/edit/5582928/